gnosis/safe-react
# refactor: rely on Onboard for provider watching

## What it solves
The removal of an unnecessary interval and, in turn, numerous requests. Also, a reduction of ENS errors in the console.

## How this PR fixes it
In researching how to reduce ENS console errors from our provider watcher, it became apparent that we were relying on a legacy method of 'watching' the Onboard provider. Onboard, however, has [built-in subscriptions](https://docs.blocknative.com/onboard#subscriptions-optional) that are a better alternative than the interval we were using. (One of which resolves ENS domains on wallet connection).

Instead of using an interval that polls every few seconds, this PR migrates to using the `wallet`, `address`, `network` and `ens` subscriptions from ENS, responsible for:

- Wallet changes
- Account changes
- Network changes
- ENS resolutions (when the above three change)

The legacy watcher interval has been removed, as well as the associated reducer/middleware adjusted. The reducer handles the subscriptions based on updates instead and wallet disconnection is now handled internally via Onboard, removing reliance on a provider removal method. Logic that was previously required in the middleware: instantiating contracts and saving the current provider for re-connection is now also handled by Onboard.

The notification system has been moved from provider 'fetching' (which was previously happening every time per interval) to the middleware, which is also subscription-based.

A few unnecessary utility functions have also be replaced or substituted for their Onboard counterparts.

## How to test it
Nothing should change on a UI/UX level. Connection may seem a bit quicker. Areas to check are the following (cross-wallet):

- Wallet connection should correctly update the `providers` store data correctly, i.e. EOA details, ENS domain, hardware/smart contract wallet flags, etc.
- Disconnection should clear the `providers` store.
- Provider-related notifications should work as before.
